include/picongpu/simulation/control/Simulation.hpp

```diff
@@ -166,26 +168,28 @@ namespace picongpu
             while(periodic.size() < 3)
                 periodic.push_back(0);
 
+
+            PMACC_VERIFY_MSG(
+                devices.size() >= 2 && devices.size() <= 3,
+                "Invalid number of devices.\nuse [-d dx=1 dy=1 dz=1]");
+
             // check on correct number of devices. fill with default value 1 for missing dimensions
-            if(devices.size() > 3)
-            {
-                std::cerr << "Invalid number of devices.\nuse [-d dx=1 dy=1 dz=1]" << std::endl;
-            }
-            else
-                while(devices.size() < 3)
-                    devices.push_back(1);
+            while(devices.size() < 3)
```

Comment:
Could you explain what you see here as a problem? This line is only filling up the vector `devices` with `1`
Line 192 is extending the `gridSize` by `1` in case we run a 2D simulation.